sound/soc/sof/intel/hda-loader.c

```diff
@@ -546,17 +546,52 @@ int hda_dsp_ipc4_load_library(struct snd_sof_dev *sdev,
 	memcpy(dmab.area, stripped_firmware.data, stripped_firmware.size);
 
 	msg.primary = hext_stream->hstream.stream_tag - 1;
-	msg.primary |= SOF_IPC4_MSG_TYPE_SET(SOF_IPC4_GLB_LOAD_LIBRARY);
+	msg.primary |= SOF_IPC4_MSG_TYPE_SET(SOF_IPC4_GLB_LOAD_LIBRARY_PREPARE);
 	msg.primary |= SOF_IPC4_MSG_DIR(SOF_IPC4_MSG_REQUEST);
 	msg.primary |= SOF_IPC4_MSG_TARGET(SOF_IPC4_FW_GEN_MSG);
-	msg.primary |= SOF_IPC4_GLB_LOAD_LIBRARY_LIB_ID(fw_lib->id);
+
+	/*
+	 * 1st stage: SOF_IPC4_GLB_LOAD_LIBRARY_PREPARE
+	 * message with valid dma_id and invalid (0) lib_id
```

Comment:
why do you even need to pass an invalid lib_id here?

Reply:
True, we only send the dma_id.